src/js/measurements.ts

```diff
@@ -0,0 +1,143 @@
+import { getMainCarrier, Hub } from "@sentry/hub";
+import { Transaction } from "@sentry/tracing";
+import { CustomSamplingContext, TransactionContext } from "@sentry/types";
+import { logger, timestampInSeconds } from "@sentry/utils";
+
+/**
+ *
+ */
+export class Stalls {
+  // private _acceptableBusyTime: number = 0;
+  private _totalStallTime: number = 0;
+  private _stallCount: number = 0;
+  private _longestStall: number = 0;
+  private _lastInterval: number = 0;
+
+  private _timeout?: number;
+
+  constructor() {
+    // this._acceptableBusyTime = 16.667;
+
+    this._lastInterval = timestampInSeconds() * 1000;
+  }
+
+  /**
+   *
+   */
+  public start(): void {
+    this._iteration();
+  }
+
+  /**
+   *
+   */
+  public finish(): {
+    stall_count: { value: number };
+    total_stall_time: { value: number };
+    longest_stall: { value: number };
+  } {
+    if (typeof this._timeout === "number") {
+      clearTimeout(this._timeout);
+    }
+
+    return {
+      stall_count: { value: this._stallCount },
+      total_stall_time: { value: this._totalStallTime },
+      longest_stall: { value: this._longestStall },
+    };
+  }
+
+  /**
+   *
+   */
+  private _iteration(): void {
+    const now = timestampInSeconds() * 1000;
+    const busyTime = now - this._lastInterval;
+
+    if (busyTime >= 80) {
+      const stallTime = busyTime - 80;
+      this._stallCount += 1;
+      this._totalStallTime += stallTime;
+      this._longestStall = Math.max(this._longestStall, stallTime);
+    }
+
+    this._lastInterval = now;
+    this._timeout = setTimeout(this._iteration.bind(this), 40);
+  }
```

Comment:
Measures the stall duration via setting a timeout to run, in this case, in 40ms, and then measures if the actual "busy time" is greater than 80ms. These numbers are just hard coded and arbitrary for now.

In theory we could measure slow and frozen frames using this by running the timeout every 16.667ms (60FPS / 1000ms) given we determine what duration constitutes "slow" frames and "frozen" frames. We should also note that running the timer every 16.667ms could have an effect on the performance of the app itself.